### PR TITLE
Additional work on top of the async-client and indexes

### PR DESCRIPTION
**Pull request by johanandren**
Mostly applying my own comments on #81

### CODE REVIEW

**Comment by johanandren** [comment]
Biggest changes are transforming `Couchbase` class and companion into a single `AsyncCouchbaseSession` trait, moving the index responsibility to the respective plugins that needs the indexes.

File: core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -40,8 +40,8 @@ couchbase-journal {
     # Fail a read from couchbase if it takes longer than this
     read-timeout = 15 s
 
-    # Parameter indicating whether the `pi2` index should be auto created if it does not exist
-    # DO NOT USE THIS IN PRODUCTION
+    # Auto create the required indexes for the write journal if they are missing when the
```
Comment (chbatey) [inline core/src/main/resources/reference.conf]: I wonder if we shouldn't do this and just rely on the global index that comes with a default couchbase install?

Comment (chbatey) [reply core/src/main/resources/reference.conf]: then we can avoid some of the init

File: core/src/main/scala/akka/persistence/couchbase/CouchbaseReadJournal.scala [hunk 2]
```diff
@@ -300,6 +325,6 @@ class CouchbaseReadJournal(eas: ExtendedActorSystem, config: Config, configPath:
 
 
    */
-  override def persistenceIds(): Source[String, NotUsed] =
-    ???
+  // override def persistenceIds(): Source[String, NotUsed] =
```
Comment (chbatey) [inline core/src/main/scala/akka/persistence/couchbase/CouchbaseReadJournal.scala]: just delete?

File: core/src/main/resources/reference.conf (hunk 1 above)
Comment (chbatey) [reply core/src/main/resources/reference.conf]: And maybe have a class they can run separately that creates the indexes that we can use in tests?

Comment (johanandren) [reply core/src/main/resources/reference.conf]: Yeah, maybe, it _is_ really useful also for just playing around with the journal with minimal setup fuzz though.

File: core/src/main/scala/akka/persistence/couchbase/CouchbaseReadJournal.scala (hunk 2 above)
Comment (johanandren) [reply core/src/main/scala/akka/persistence/couchbase/CouchbaseReadJournal.scala]: We'll implement it next week or so, I just thought it was unfair to claim we support it by implementing the trait with a stub.

**Pull request merged**